hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -283,7 +283,7 @@ public abstract class Dialect implements ConversionContext, TypeContributor, Fun
 	private static final Pattern ESCAPE_CLOSING_COMMENT_PATTERN = Pattern.compile( "\\*/" );
 	private static final Pattern ESCAPE_OPENING_COMMENT_PATTERN = Pattern.compile( "/\\*" );
 	private static final Pattern QUERY_PATTERN = Pattern.compile(
-		"^\\s*(select\\b.+?\\bfrom\\b.+?)(\\b(?:natural )?(?:left |right |full )?(?:inner |outer |cross )?join.+?\\b)?(\\bwhere\\b.+?)$");
+		"^\\s*(select\\b.+?\\bfrom\\b.+?)(\\b(?:(?:natural )?(?:left |right |full )?(?:inner |outer |cross )?join |straight_join).+?\\b)?(\\bwhere\\b.+?)$");
```

Comment:
It looks like the pattern might benefit from being more explicit about necessary whitespace:

```suggestion
		"^\\s*(select\\s.+?\\sfrom\\s.+?)(\\s(?:(?:natural\\s+)?(?:left\\s+|right\\s+|full\\s+)?(?:inner\\s|outer\\s|cross\\s)?join\\s|straight_join\\s).+?\\b)?(\\swhere\\s.+?)$");
```

Reply:
Thank you for the suggestion.
You're right; the original regex can't handle some queries with new lines.

Based on your suggestion, I've modified the regex as follows:
1. `\\b` -> `\\s` for improved robustness
2. `(?:A\\s|B\\s|C\\s)?` -> `(?:A|B|C)?\\s*` for better readability